packages/core/src/extensions/DraggableBlocks/DraggableBlocksPlugin.ts

```diff
@@ -237,14 +238,13 @@ export class BlockMenuView {
   // When false, the drag handle with be just to the left of the element
   horizontalPosAnchoredAtRoot: boolean;
 
-  blockMenuParams: BlockSideMenuParams;
   blockMenu: BlockSideMenu;
 
+  hoveredBlock: HTMLElement | undefined;
```

Comment:
I'm not sure how safe this is. We call `getBoundingClientRect` at a moment that we're not sure thism element still exists. Consider the following scenario:

- hover element
- click draghandle
- a different plugin, let's say a spellchecker or whatsoever, changes an attribute of the block
-  now we click "add"

I think this won't work, as the htmlelement of the block can have changed

Reply:
I think the safest way to do it would be to save a hoveredBlockId instead and each time we need to get the block HTML element & bounding box for a function we could call:
```
document.querySelectorAll('[data-id=this.hoveredBlockId]')
```
This would imo be relatively clean but pretty inefficient.

We could also save the hovered block's bounding box instead which would be nicer and more efficient, but could still cause problems if another extension changes e.g. the block type in the meantime.

Any thoughts on which solution is better?

Reply:
Maybe the easiest is to stick to the current solution, but use your `document.querySelectorAll('[data-id=this.hoveredBlockId]')` idea if `hoveredBlockId.isConnected` is false?